lib/schemas.py

```diff
@@ -1,5 +1,8 @@
 from pydantic import BaseModel, ValidationError
 from typing import Any, Dict, List, Optional, Union
+class ErrorResponse(BaseModel):
+    error: Optional[str] = None
+    error_details: Optional[Dict] = None
 
 class MediaResponse(BaseModel):
     hash_value: Optional[Any] = None
```

Comment:
Maybe there's something else we should be doing here - we should only be rescuing for "acceptable" errors with this method, but allowing other ones to properly explode the normal way? And then in that case, it's up to each model to narrowly define the subset of acceptable failure modes?

Reply:
It makes sense to me that the model defines what it considers errors it wants to send back.   And Presto is providing the template for "this is how you must structure an error response payload" so we don't have to process it differently for every model.   @ashkankzme  is this the use case you were looking for?

Reply:
Yeah - ultimately I'm happy with whatever here - I see an argument for a catch-all implementation where you just let this level absorb, log, handle the error, but also see the point in having custom returns for the narrowly acceptable band of enumerated errors...

Reply:
what Devin has implemented here is pretty much what I was looking for. 
Depending on the types of errors the consumers of Presto may want to act differently (e.g. retry, not retry, call a backup service, etc) to standardize the error responses. 

To Skye's point, what if `ErrorResponse` had an `error_status_code` field that mirrored the http status codes (400 range for most cases), and that should be set by whatever Presto endpoint that is raising the exception, or else it would resort to some default code, 400 or something?
Having this would require a base exception class that includes the status code, but other than that it should be ok?

Reply:
How do you feel about @ashkankzme's suggestion here? Perhaps we amend the schema as:

```
class ErrorResponse(BaseModel):
    error: Optional[str] = None
    error_details: Optional[Dict] = None
    error_code: Optional[int] = 500
```

And later we can add in some capacity for setting that from within the process attempt itself?

Reply:
this seems fine (as long as error code is not optional).   However, the field that determines 'is this an error or not' needs to be in the BaseModel right? otherwise how do we know if we will be expecting to parse error fields?

Reply:
I disagree that it needs to be in `BaseModel`. The `Message` structure is a hierarchical series of envelopes containing, at the top `Message` level, the routing information, at the middle `GenericItem` level the details about the task to be completed, and at the `*Result` level the resulting output of the attempt to complete the task. I think the appropriate level to contain the error, as a result, as at that innermost level, which will be returned as JSON to any callback event. Does that help clarify what's going on conceptually here?

Reply:
> I disagree that it needs to be in BaseModel

Proposal:  validate the design by including a test that receives a callback and acts conditionally by checking if it was an error or not

Reply:
`GenericItem` would have a `result`, which in the case of an error, would be of type `ErrorResponse` which is how you could check that it was an errored case. You'd send along the entire message body, but embedded in the `result` would be an unambiguous signal that the result was an error. Is that sufficient?